FindChannelFunction/src/main/java/no/unit/nva/channel/FindChannelFunctionApp.java

```diff
@@ -0,0 +1,120 @@
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.apache.http.HttpStatus.SC_NOT_FOUND;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+import static org.zalando.problem.Status.BAD_REQUEST;
+import static org.zalando.problem.Status.INTERNAL_SERVER_ERROR;
+import static org.zalando.problem.Status.NOT_FOUND;
+
+public class FindChannelFunctionApp implements RequestStreamHandler {
+
+    private final transient ObjectMapper objectMapper;
+    private final transient ChannelRegistryClient channelRegistryClient;
+
+    public static final String CHANNEL_REGISTRY_URI = "https://api.nsd.no/dbhapitjener/Tabeller/hentJSONTabellData";
+    public static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+    public static final String PROBLEM_JSON = "application/problem+json";
+    public transient String allowedOrigin;
+
+    public FindChannelFunctionApp() {
+        this(createObjectMapper(), new ChannelRegistryClient(createObjectMapper(), HttpClients.createDefault(),
+                CHANNEL_REGISTRY_URI), new Environment());
+    }
+
+    public FindChannelFunctionApp(ObjectMapper objectMapper, ChannelRegistryClient channelRegistryClient, Environment environment) {
+        this.objectMapper = objectMapper;
+        this.channelRegistryClient = channelRegistryClient;
+        this.allowedOrigin = environment.get("ALLOWED_ORIGIN").orElseThrow(IllegalStateException::new);
+    }
+
+    @Override
+    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
+        SearchRequest searchRequest;
+        try {
+            JsonNode event = objectMapper.readTree(input);
+            searchRequest = objectMapper.readValue(event.get("body").asText(), SearchRequest.class);
+        } catch (Exception e) {
+            writeErrorResponse(output, Problem.valueOf(BAD_REQUEST, e.getMessage(), createProblemInstance(context)),
+                    SC_BAD_REQUEST);
+            return;
+        }
+
+        try {
+            List<Channel> channels = channelRegistryClient.fetchChannels(searchRequest.getTableId(),
+                    searchRequest.getSearchTerm());
+            SearchResponse searchResponse = new SearchResponse(channels);
+            writeResponse(output, searchResponse, SC_OK);
+        } catch (NoResultsFoundException e) {
+            writeErrorResponse(output, Problem.valueOf(NOT_FOUND, e.getMessage(), createProblemInstance(context)),
+                    SC_NOT_FOUND);
+        } catch (Exception e) {
+            System.out.println(e);
+            writeErrorResponse(output, Problem.valueOf(INTERNAL_SERVER_ERROR, e.getMessage(),
+                    createProblemInstance(context)), SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    private void writeResponse(OutputStream output, Object body, int statusCode) throws IOException {
```

Comment:
At the present time, statusCode is always 200, however, do we see any changes in this? I doubt it, but worth checking